### ISSUE TITLE
Add support for shellcheck source-path=SCRIPTDIR

### ISSUE THREAD

**Issue by mickaelperrin** [BashSupport-Pro/bashsupport-pro#187]
Hi,

I see that some work has been made to have a partial support of the `source-path` directive in #141.

However, I am unable to have it working as explained in (https://github.com/koalaman/shellcheck/wiki/Directive#source-path) the shellcheck documentation:

```
#!/bin/sh
# shellcheck source-path=SCRIPTDIR
here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
. "$here/config.sh"
. "$here/utils.sh"
```

In that example, both sourced files would trigger the `BashSupport Pro can't follow a non-constant source.` error.

Hope it will be introduced soon 👍

**Comment by jansorg** [comment]
Thanks for reporting!
The problem is not the missing support for `source-path=`, but the variable in the sourced path (`$here/config.sh`).
ShellCheck drops the first dynamic path segment of such path, but BashSupport Pro currently does not.

I'll implement it in BashSupport Pro in the same way, but I can't tell yet if that's going to land in 5.x or an update of 4.x.

This would work, but it's probably not suitable for your use-case:
```bash
#!/bin/sh
# shellcheck source-path=SCRIPTDIR
here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
cd "$here"
. "config.sh"

**Comment by jansorg** [comment]
@mickaelperrin I've fixed this for the next update of 4.x, which should be out early November.
I've changed the implementation to follow ShellCheck.

This also covers a `. "$here/file.bash"` without a ShellCheck directive. ShellCheck seems to always drop a prefix consisting of a dynamic value. The above would resolve to `./file.bash`.

**Comment by jansorg** [comment]
@mickaelperrin I've released 4.4.0 with the fix. Please let me know if the feature is not working as expected. Thanks for reporting this!
